woodwork/table_accessor.py

```diff
@@ -129,6 +129,21 @@ def schema(self):
         if self._schema:
             return self._schema._get_subset_schema(list(self.columns.keys()))
 
+    def set_index(self, new_index):
```

Comment:
I'm not sure that we need an index `@setter`. Previously, the difference between `DataTable.set_index` and the setter was that the setter made the change inplace while `set_index` created a new DataTable. We've been sticking with inplace changes so far, so it seems redundant to me to have both.

My reasoning for going with `set_index` over the setter was that we'd need to add the getter as well, and it already exists on the Schema.

This thought process also lead me to discover that the `@property` tag doesn't propagate with the way we get Schema attrs through the Accessor. I documented this in #602

Reply:
Sounds good!